proxy-node-gateway/src/test/java/uk/gov/ida/notification/session/TestRedisSessions.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.ida.notification.session;
+
+import com.github.fppt.jedismock.RedisServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import uk.gov.ida.notification.redis.SessionCacheManager;
+import uk.gov.ida.notification.exceptions.SessionAlreadyExistsException;
+import uk.gov.ida.notification.exceptions.SessionMissingException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestRedisSessions {
+
+    private static RedisServer server = null;
+    private SessionCacheManager sessionCacheManager = null;
+
+    @Mock
+    private GatewaySessionData testData;
+
+    private static final String TEST_KEY = "TEST_KEY";
+
+    @Before
+    public void before() throws IOException, URISyntaxException {
+        server = RedisServer.newRedisServer();
+        server.start();
```

Comment:
Do you trust your library? 😁 If you do, you don't need to test that lettuce interacts with Redis correctly, all you need to do is test that you are using lettuce correctly. [The hub](https://github.com/alphagov/verify-hub/blob/master/hub/policy/src/test/java/uk/gov/ida/hub/policy/session/RedisSessionStoreTest.java) doesn't spin up Redis at all for unit testing the session store, it just mocks redis out and verifies the correct methods were called.